app/src/organisms/ProtocolSetupLabware/index.tsx

```diff
@@ -0,0 +1,481 @@
+          return mostRecentAnalysis != null ? (
+            <RowLabware
+              key={i}
+              labware={labware}
+              robotSideAnalysis={mostRecentAnalysis}
+              attachedProtocolModules={attachedProtocolModuleMatches}
+              refetchModules={moduleQuery.refetch}
+            />
+          ) : null
+        })}
+      </Flex>
+      <DeckMapButton onClick={() => setShowDeckMapModal(true)} />
+    </>
+  )
+}
+
+interface LabwareLatchProps {
+  toggleLatch: () => void
+  latchStatus: LatchStatus
+}
+
+function LabwareLatch({
+  toggleLatch,
+  latchStatus,
+}: LabwareLatchProps): JSX.Element {
+  const { t } = useTranslation(['protocol_setup', 'heater_shaker'])
+  let buttonState: JSX.Element | null = null
+  switch (latchStatus) {
+    case 'idle_open':
+      buttonState = (
+        <Flex
+          width="100%"
+          justifyContent={JUSTIFY_SPACE_BETWEEN}
+          alignItems={ALIGN_CENTER}
+        >
+          <StyledText
+            fontSize="1.375rem"
+            fontWeight={TYPOGRAPHY.fontWeightRegular}
+          >
+            {t('heater_shaker:open')}
+          </StyledText>
+          <Icon name="latch-open" size="2.5rem" color="#006CFA" />
+        </Flex>
+      )
+      break
+    case 'idle_closed':
+      buttonState = (
+        <Flex
+          width="100%"
+          justifyContent={JUSTIFY_SPACE_BETWEEN}
+          alignItems={ALIGN_CENTER}
+        >
+          <StyledText
+            fontSize="1.375rem"
+            fontWeight={TYPOGRAPHY.fontWeightRegular}
+          >
+            {t('heater_shaker:closed')}
+          </StyledText>
+          <Icon name="latch-closed" size="2.5rem" />
+        </Flex>
+      )
+      break
+    case 'opening':
+      buttonState = (
+        <Flex
+          width="100%"
+          justifyContent={JUSTIFY_SPACE_BETWEEN}
+          alignItems={ALIGN_CENTER}
+        >
+          <StyledText
+            fontSize="1.375rem"
+            fontWeight={TYPOGRAPHY.fontWeightRegular}
+          >
+            {t('heater_shaker:opening')}
+          </StyledText>
+          <Icon name="latch-closed" size="2.5rem" />
+        </Flex>
+      )
+      break
+    case 'closing':
+      buttonState = (
+        <Flex
+          width="100%"
+          justifyContent={JUSTIFY_SPACE_BETWEEN}
+          alignItems={ALIGN_CENTER}
+        >
+          <StyledText
+            fontSize="1.375rem"
+            fontWeight={TYPOGRAPHY.fontWeightRegular}
+          >
+            {t('heater_shaker:closing')}
+          </StyledText>
+          <Icon name="latch-open" size="2.5rem" />
+        </Flex>
+      )
+      break
+    default:
+      buttonState = null
+  }
+  return (
+    <Flex
+      minWidth="11.4375rem"
+      height="7rem"
+      flexDirection={DIRECTION_COLUMN}
+      justifyContent={JUSTIFY_SPACE_BETWEEN}
+      alignItems={ALIGN_FLEX_START}
+      padding={SPACING.spacing4}
+      gridGap="0.75rem"
+      backgroundColor="#B4D4FF"
```

Comment:
```suggestion
      backgroundColor={COLORS. foundationalBlue}
```
maybe need to merge edge into your branch.